src/App.js

```diff
@@ -11,88 +11,127 @@ const GAME_SPEED = 3000;
-  const [transplant, setTransplant] = useState(INIT.transplant);
-  const [ponds, setPonds] = useState(INIT.ponds);
+  const [production, setProduction] = useState(INIT.production);
 
+  /**
+   * Can be used for 
+   */
   const gameLoop = () => {
-    cyclePonds();
+    cycleProduction();
     resetTimer();
     setDay((day) => day + 1);
-    console.log("tick");
   };
 
+  // A bit of a hack to reset the css animation on the timer
   const resetTimer = () => {
-    // A bit of a hack to reset the animation on the timer
     setTimer(false);
     setTimer(true);
   };
 
-  // Empty out the tranplant array and add each into a separate pond queue
-  const cyclePonds = () => {
-    let transplants = [];
-    setTransplant((transplant) => {
-      transplants = _.cloneDeep(transplant);
-      return INIT.transplant;
-    });
-    setPonds((ponds) => {
-      const newPonds = _.cloneDeep(ponds);
-      newPonds.forEach((pond, i) => {
-        const raft = transplants.rafts[i];
-        pond.rafts.unshift(raft);
-        pond.rafts.pop();
+  /**
+   * Adds a new raft to each pond in production
+   */ 
+  const cycleProduction = () => {
+    setProduction((production) => {
+      const newProduction = _.cloneDeep(production);
+      newProduction.forEach((pond) => {
+        pond.rafts.unshift(new ProductionRaft());
       });
-      return newPonds;
+      return newProduction;
     });
   };
 
-  const start = () => {
-    setIntervalId(setInterval(gameLoop, GAME_SPEED));
-    setTimer(true);
-  };
-
-  const reset = () => {
-    clearInterval(intervalId);
-    setDay(INIT.day);
-    setGermination(INIT.germination);
-    setTransplant(INIT.transplant);
-    setPonds(INIT.ponds);
-    setTimer(false);
-  };
-
+  /**
+   * Logs the game's data objects to console
+   */
   const debug = () => {
     console.log({
       day,
       germination,
-      transplant,
-      ponds,
+      production,
       timer
     });
   };
 
-  const addRaft = (pondIndex) => {
-    const newTransplant = _.cloneDeep(transplant);
-    newTransplant.rafts[pondIndex] = new Raft();
-    setTransplant(newTransplant);
+  /**
+   * Transplants seedlings from the oldest seeded tray to the newest raft of the 
+   * selected pond.
+   * @param {number} pondIndex the index of the pond to transplant to
+   */
+
+  const transplantToPond = (pondIndex) => {
+    const germinationCopy = _.cloneDeep(germination);
+    const productionCopy = _.cloneDeep(production);
+
+    const sourceTray = germinationCopy.trays.pop();
+    const destinationRaft = productionCopy[pondIndex].rafts[0];
+    
+    const plantsRequested = destinationRaft.open;
+    const plantsAvailable = sourceTray.plants;
+    const plantsUsed = Math.min(plantsRequested, plantsAvailable);
+
+    sourceTray.plants -= plantsUsed;
+    destinationRaft.plants += plantsUsed;
+
+    if(sourceTray.plants > 0) {
+      germinationCopy.trays.push(sourceTray);
+    }
```

Comment:
this feels like a good thing to leave out as an exercise to the interviewee. So far (in my experience) it's been a bit of a stumbling block, which may or may not be a good thing. We can add it for now and see how it plays out but maybe consider removing it in the future?

Reply:
sounds good 👍